sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -5664,6 +5664,20 @@ pi_result piextEventCreateWithNativeHandle(pi_native_handle NativeHandle,
   // need to create a host-visible proxy for it.
   (*Event)->HostVisibleEvent = *Event;
 
+  // Regular events managed by SYCL RT are created with RefCnt == 2 so that
+  // their lifetime is extended beyond that of of the associated pi_event as we
+  // need underlying ZeEvent available for the driver to write back to once
+  // device code finishes.
+  //
+  // There is no such requirement for the events constructed from the native
+  // ZeEvent. Indeed, it isn't even mandatory for that event to ever be
+  // signalled/associated with device task/etc. As such, the runtime has no
+  // knowledge or responsibility to extened lifetime of the underlying ZeEvent.
+  // Instead, it's the user's responsibility to ensure that the lifetime of
+  // sycl::event created from the native handle (if the ownership is being
+  // transferred) is such that its lifetime doesn't end prematurely.
+  (*Event)->CleanedUp = true;
```

Comment:
I understand that you have it here to avoid the call to piEventRelease made in cleanup() to match the retain that we make for regular SYCL events. But the event cleanup() call that would not be made with this change processes release of dependent events (which are probably none here) and some other stuff. Is it all not necessary? 

@againull : what do you think?

Reply:
I can't think of any scenario where interop event has dependent events, associated queue or associated kernel. So I believe it is ok that we don't call cleanup for this event.

Reply:
OK, let's go with it.